hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SBaseStruct.scala

```diff
@@ -57,6 +57,23 @@ trait SBaseStructValue extends SValue {
     new SInt32Value(hash_result)
   }
 
+  override def sizeToStoreInBytes(cb: EmitCodeBuilder): SInt64Value = {
+    // Size in bytes of the struct that must represent this thing, plus recursive call on any non-missing children.
+    val pStructSize = this.st.storageType().byteSize
+    val sizeSoFar = cb.newLocal[Long]("sstackstruct_size_in_bytes", pStructSize)
+    (0 until st.size).foreach { idx =>
+      if (!this.st.fieldTypes(idx).isPrimitive) {
```

Comment:
I think there are non-primitive types that are stored inline, so I think that we should switch on if the value points outside the struct, rather than if it's primitive.

Reply:
Hmm, do we have a way to check if the value points outside the struct?

Reply:
`containsPointers` check if something has pointers inside it, but it doesn't tell me if a particular element of this struct is inline or not.